services/src/main/java/org/exoplatform/addons/gamification/listener/social/profile/GamificationProfileListener.java

```diff
@@ -112,5 +127,17 @@ public void aboutMeUpdated(ProfileLifeCycleEvent event) {
                 event.getProfile().getUrl());
     }
 
+    private void clearUserActivitiesCache(String userId) {
+      List<Announcement> announcements = announcementService.getAnnouncementsByEarnerId(userId);
+      if (CollectionUtils.isEmpty(announcements)) {
+        return;
+      }
+      announcements.forEach(announcement -> clearCache(String.valueOf(announcement.getActivityId())));
+    }
 
+    private void clearCache(String activityId) {
+      if (cachedActivityStorage != null) {
+        cachedActivityStorage.clearActivityCached(activityId);
+      }
```

Comment:
if `CachedActivityStorage` is added in constructor as dependency, then it can't be null. Besides, it's may be better to add this methods in Service Layer in ActivityManager API.

BTW, it was already treated in Kudos here: https://github.com/Meeds-io/kudos/pull/287/files if this helps as well to use the same behavior.

```suggestion
      cachedActivityStorage.clearActivityCached(activityId);
```

Reply:
Updated